core/Loc.cc

```diff
@@ -57,6 +57,12 @@ optional<uint32_t> Loc::pos2Offset(const File &file, Loc::Detail pos) {
         return nullopt;
     }
     auto lineOffset = lineBreaks[l];
+    // TODO(jez) Should be easy to add a before+after test for this (wrap around)
+    auto nextLineStart = l + 1 < lineBreaks.size() ? lineBreaks[l + 1] : file.source().size();
+    auto lineLength = nextLineStart - lineOffset;
+    if (pos.column > lineLength) {
+        return nullopt;
+    }
```

Comment:
There are some edge cases here:

- position is start of newline character
- position is end of newline character
- position is end of buffer

please don't let me merge this until these are tested.